lib/resources/payment_method.ex

```diff
@@ -0,0 +1,109 @@
+defmodule Iugu.PaymentMethod do
+  @moduledoc """
+  Iugu's payment method resource
+  """
+
+  defstruct ~w(id description item_type data customer_id)a
+
+  @spec list(%Iugu.Customer{}) :: Iugu.Request.get_response()
+  @doc """
+  List Iugu's Customer Payment Methods
+
+  ## Parameters
+    - customer_id: `%Iugu.Customer{}` or a string of Iugu Customer ID
+  """
+  def list(%Iugu.Customer{id: id}) do
+    list(id)
+  end
+
+  @spec list(String.t()) :: Iugu.Request.get_response()
+  def list(customer_id) do
+    %Iugu.Request{path: "customers/#{customer_id}/payment_methods"}
+    |> Iugu.Request.get(__MODULE__, :collection)
+  end
+
+  @spec create(%Iugu.Customer{}, map) :: Iugu.Request.post_response()
+  @doc """
+  Creates a Iugu's Payment Method to customer
+
+  ## Parameters
+    - customer_id: `%Iugu.Customer{}` or a string of Iugu Customer ID
+    - data: map with attributes - [Iugu Reference](https://dev.iugu.com/v1.0/reference#testinput-3)
+  """
+  def create(%Iugu.Customer{id: id}, %{} = data) do
+    create(id, data)
+  end
+
+  @spec create(String.t(), map) :: Iugu.Request.post_response()
+  def create(customer_id, %{} = data) do
```

Comment:
One of the things that I would like to bring to the contract of actions is a way to also receive the request's data (`%Iugu.Request{}`) as already done for the [list action](https://github.com/tiagopog/iugu-elixir/blob/master/lib/base/resource.ex#L27-L36).

For example, having this quoted expression for the `create` action on `Iugu.Resource`:

```elixir
  def define_action(:create) do
    quote do
      @spec create(map, Iugu.Request.t()) :: Iugu.Request.post_response
      def create(%{} = data, request \\ %Iugu.Request{}) do
        case Poison.encode(data) do
          {:ok, json} ->
            %Iugu.Request{path: @resource, body: json}
            |> Map.merge(request)
            |> Iugu.Request.post(__MODULE__)

          {status, result} ->
            {status, result}
        end
      end
    end
  end
```

Then we could simplify things a bit:

```elixir
def create(customer_id, %{} = data) do
  create(data, %Iugu.Request{path: "customers/#{customer_id}/payment_methods"})
end
```

This could be applied for the actions as well. What do you think?